cdc/puller/ddl_puller.go

```diff
@@ -270,37 +270,40 @@ func (p *ddlJobPullerImpl) handleRenameTables(job *timodel.Job) (skip bool, err
 	remainTables := make([]*timodel.TableInfo, 0, len(multiTableInfos))
 	snap := p.schemaStorage.GetLastSnapshot()
 	for i, tableInfo := range multiTableInfos {
-		schema, ok := snap.SchemaByID(newSchemaIDs[i])
+		var shouldDiscardNewTable, shouldDiscardOldTable bool
+		newSchemaName, ok := snap.SchemaByID(newSchemaIDs[i])
 		if !ok {
-			return true, cerror.ErrSnapshotSchemaNotFound.GenWithStackByArgs(newSchemaIDs[i])
```

Comment:
// can't find the new table schema, means the new table name does not hit the filter rule and we should discard the new table